packages/xstate-solid/src/types.ts

```diff
@@ -0,0 +1,53 @@
+import type { EventObject, StateConfig } from 'xstate';
+import {
+  AnyStateMachine,
+  AreAllImplementationsAssumedToBeProvided,
+  InternalMachineOptions,
+  InterpreterFrom,
+  InterpreterOptions,
+  StateFrom
+} from 'xstate';
+
+interface UseMachineOptions<TContext, TEvent extends EventObject> {
+  /**
+   * If provided, will be merged with machine's `context`.
+   */
+  context?: Partial<TContext>;
+  /**
+   * The state to rehydrate the machine to. The machine will
+   * start at this state instead of its `initialState`.
+   */
+  state?: StateConfig<TContext, TEvent>;
+}
+
+type Prop<T, K> = K extends keyof T ? T[K] : never;
+
+export type UseMachineReturn<
+  TMachine extends AnyStateMachine,
+  TInterpreter = InterpreterFrom<TMachine>
+> = [StateFrom<TMachine>, Prop<TInterpreter, 'send'>, TInterpreter];
+
+export type RestParams<
```

Comment:
note for other reviewers - this is the same here between `createService` and `useMachine` because `createService` doesn't accept `observerOrListener` in this package (which I think is fine)